.github/workflows/freertos_plus_demos.yml

```diff
@@ -1051,10 +1061,19 @@ jobs:
           echo "::endgroup::"
           echo -e "${{ env.bashPass }} ${{ env.stepName }} ${{ env.bashEnd }} "
 
+      - name: Query device IP address
+        id: get-ip
+        run: |
+          # Query the device's IP address
+          SERVER_IP="$(ip addr show | grep "inet " | grep -v "127.0.0.1" | awk '{print $2}' | cut -d/ -f1 | head -n 1)"
+          echo "Device IP address: $SERVER_IP"
+          echo "server_ip=\"$SERVER_IP\"" >> $GITHUB_ENV
+
       - name: Start localhost Echo server
         id: echo-server
-        uses: FreeRTOS/CI-CD-GitHub-Actions/localhost-echo-server@main
+        uses: tony-josi-aws/CI-CD-GitHub-Actions/localhost-echo-server@main
```

Comment:
`tony-josi-aws/CI-CD-GitHub-Actions/localhost-mqtt-broker@main` is used for debug